### PR TITLE
Update README to be ergonomic and excite new users

### PR DESCRIPTION
**Pull request by sungchun12**
The front page of the open source tool needs some refinement and polish to entice new users and show the world we mean serious business!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,79 +9,61 @@ data-diff: Compare datasets fast, within or across SQL databases
...
-## Data Migration & Replication Testing
-Compare source to target and check for discrepancies when moving data between systems:
-- Migrating to a new data warehouse (e.g., Oracle > Snowflake)
-- Converting SQL to a new transformation framework (e.g., stored procedures > dbt)
-- Continuously replicating data from an OLTP DB to OLAP DWH (e.g., MySQL > Redshift)
+# Use Cases
 
+### Data Migration & Replication Testing
+data-diff is a powerful tool for comparing data when you're moving it between systems. Use it to ensure data accuracy and identify discrepancies during tasks like:
+- **Migrating** to a new data warehouse (e.g., Oracle > Snowflake)
```
Comment (sungchun12) [inline README.md:23]:
```suggestion
- **Migrating** to a new data warehouse (e.g., Oracle -> Snowflake)
```

File: README.md [hunk 2]
```diff
@@ -9,79 +9,61 @@ data-diff: Compare datasets fast, within or across SQL databases
...
+- **Converting SQL** to a new transformation framework (e.g., stored procedures > dbt)
```
Comment (sungchun12) [inline README.md:24]:
```suggestion
- **Converting SQL** to a new transformation framework (e.g., stored procedures -> dbt)
```

File: README.md [hunk 3]
```diff
@@ -9,79 +9,61 @@ data-diff: Compare datasets fast, within or across SQL databases
...
+- Continuously **replicating data** from an OLTP database to OLAP data warehouse (e.g., MySQL > Redshift)
```
Comment (sungchun12) [inline README.md:25]:
```suggestion
- Continuously **replicating data** from an OLTP database to OLAP data warehouse (e.g., MySQL -> Redshift)
```

File: README.md [hunk 4]
```diff
@@ -189,9 +178,21 @@ Your database not listed here?
 
 <br>
 
+# How it works
+
+`data-diff` efficiently compares data using two modes:
+
+1. **`joindiff`**: Ideal for comparing data within the same database, utilizing outer joins for efficient row comparisons. It relies on the database engine for computation and has consistent performance.
```
Comment (sungchun12) [inline README.md:185]: Can you convert these into toggles for both joindiff and hashdiff? Let's not force readers to context switch for a quick explainer. I agree with moving the how it works section here. 

## `joindiff`
- Recommended for comparing data within the same database
- Uses the outer join operation to diff the rows as efficiently as possible within the same database
- Fully relies on the underlying database engine for computation
- Requires both datasets to be queryable with a single SQL query
- Time complexity approximates JOIN operation and is largely independent of the number of differences in the dataset

## `hashdiff`
- Recommended for comparing datasets across different databases
- Can also be helpful in diffing very large tables with few expected differences within the same database
- Employs a divide-and-conquer algorithm based on hashing and binary search
- Can diff data across distinct database engines, e.g., PostgreSQL <> Snowflake
- Time complexity approximates COUNT(*) operation when there are few differences
- Performance degrades when datasets have a large number of differences

**Review by sungchun12 — COMMENTED** [review]
Made a couple edits directly, made suggestions where relevant, add that portion on open source vs. datafold cloud and we're good to go!

File: README.md (hunk 4 above)
Comment (sungchun12) [reply README.md:185]: Keep your new summary sentences though.

File: README.md [hunk 5]
```diff
@@ -9,79 +9,61 @@ data-diff: Compare datasets fast, within or across SQL databases
...
-- Time complexity approximates COUNT(*) operation when there are few differences
-- Performance degrades when datasets have a large number of differences
+### ⚡ Validating dbt model changes between dev and prod
+Looking to use data-diff in dbt development? 
 
-More information about the algorithm and performance considerations can be found [here](https://github.com/datafold/data-diff/blob/master/docs/technical-explanation.md)
+Development testing with Datafold enables you to see the impact of dbt code changes on data as you write the code, whether in your IDE or CLI. 
 
-# Get started
+ Head over to [our `data-diff` + `dbt` documentation](https://docs.datafold.com/development_testing/how_it_works) to get started with a development testing workflow!
```
Comment (sungchun12) [inline README.md:53]:
```suggestion
 Head over to [our `data-diff` + `dbt` documentation](https://docs.datafold.com/development_testing/cli/) to get started with a development testing workflow!
```

File: README.md (hunk 1 above)
Comment (elliotgunn) [reply README.md:23]: fixed

File: README.md (hunk 2 above)
Comment (elliotgunn) [reply README.md:24]: fixed

File: README.md (hunk 3 above)
Comment (elliotgunn) [reply README.md:25]: fixed

File: README.md (hunk 4 above)
Comment (elliotgunn) [reply README.md:185]: That's a good option - did not think of toggles!

Comment (elliotgunn) [reply README.md:185]: re-added

File: README.md (hunk 5 above)
Comment (elliotgunn) [reply README.md:53]: fixed

**Review by elliotgunn — COMMENTED** [review]
thanks @sungchun12 for the edits, we're ready to merge

**Review by elliotgunn — APPROVED** [review]
forgot to approve since I'm the reviewer

File: README.md [hunk 6]
```diff
@@ -148,11 +134,18 @@ data-diff --conf datadiff.toml \
...
 ```
 
+5. Review the output
+
+After running your data-diff job, review the output to identify and analyze differences in your data.
+
 Check out [documentation](https://docs.datafold.com/reference/open_source/cli) for the full command reference.
 
+# data-diff OSS & Datafold Cloud
+data-diff is an open source utility. You run stateless diffs on your local computer but results cannot be easily accessed by others.
```
Comment (sungchun12) [inline README.md:144]:
```suggestion
data-diff is an open source utility for running stateless diffs on your local computer for a great single player experience.
```

File: README.md [hunk 7]
```diff
@@ -148,11 +134,18 @@ data-diff --conf datadiff.toml \
 + 1, returned
 ```
 
+5. Review the output
+
+After running your data-diff job, review the output to identify and analyze differences in your data.
+
 Check out [documentation](https://docs.datafold.com/reference/open_source/cli) for the full command reference.
 
+# data-diff OSS & Datafold Cloud
```
Comment (sungchun12) [inline README.md:143]: Let's move this section right above the contributions section. This doesn't narratively make sense to inject here.

Comment (elliotgunn) [reply README.md]: fixed

**Pull request merged**